core/domain/suggestion_services.py

```diff
@@ -255,9 +258,38 @@ def reject_suggestion(suggestion, reviewer_id, review_message):
     mark_review_completed(
         suggestion, suggestion_models.STATUS_REJECTED, reviewer_id)
 
+    thread_id = suggestion.suggestion_id
+    if not constants.ENABLE_GENERALIZED_FEEDBACK_THREADS:
+        thread_id = thread_id[thread_id.find('.') + 1:]
     feedback_services.create_message(
-        suggestion.suggestion_id, reviewer_id,
-        feedback_models.STATUS_CHOICES_IGNORED, None, review_message)
+        thread_id, reviewer_id, feedback_models.STATUS_CHOICES_IGNORED,
+        None, review_message)
+
+
+def get_all_suggestions_that_can_be_reviewed_by_user(user_id):
+    """Returns a list of suggestions which need to be reviewed, in categories
+    where the user has crossed the minimum score to review.
+
+    Args:
+        user_id: str. The ID of the user.
+
+    Returns:
+        list(Suggestion). A list of suggestions which the given user is allowed
+            to review.
+    """
+    scores = get_all_scores_of_user(user_id)
+    score_categories = []
+    for score_category, score in scores.items():
```

Comment:
@nithusha21 can we do this in the model itself with the query? Generally, it is a good practice to filter the result set in the query itself.